### PR TITLE
(PDB-1177) add noop flag to reports

### PR DESCRIPTION
**Pull request by wkalt**
This adds a "noop" field to the reports object, which is a boolean flag
indicating whether the run producing the report was a --noop.

### CODE REVIEW

**Comment by wkalt** [comment]
@pljenkinsro retest this please

**Comment by wkalt** [comment]
@pljenkinsro retest this please

**Comment by wkalt** [comment]
@pljenkinsro retest this please

File: acceptance/tests/reports/report_storage.rb [hunk 1]
```diff
@@ -2,14 +2,14 @@
 
 test_name "basic validation of puppet report submission" do
 
-  step "setup a test manifest for the master and perform agent runs" do
-    manifest = <<-MANIFEST
+  manifest = <<-MANIFEST
```
Comment (kbarber) [inline acceptance/tests/reports/report_storage.rb]: I'm not sure we should be overloading noop related testing with an existing test here. I think a new test is in order.

Comment (kbarber) [inline acceptance/tests/reports/report_storage.rb]: Although I can see what you've done at the end of this file. Somehow I think this should be inverted though, we've gone and modified an existing test with noop stuff, and then added back a non-noop test. I'd at least invert the way we're doing this here, the noop test should be the bit we've added at the end, not the other way around.

**Pull request merged**